Change utility scripts to account for architecture-aware CFN mappings

tilne opened the pull request:
* Most important change is `generate-ami-list.py`, which we use as part of both the testing and release process
* I figured it best to also update `get-ami-list.py` and `ami_convert.py`, but from what I could see we don't use these scripts anywhere?

`amis.txt` and the main CFN template are also modified. The intent is to show the changes to the format, and those changes will be backed out before merging to develop.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

rexcsn commented on util/generate-ami-list.py:
```diff
@@ -189,19 +228,23 @@ def update_cfn_template(cfn_template_file, amis_to_update):
+    current_amis = get_initialized_mappings_dicts()
+    for mapping_name, regions in current_amis.items():
+        current_amis_for_mapping = cfn_data.get("Mappings").get(mapping_name, {})
+        for region, os_to_amis in current_amis_for_mapping.items():
```
Could we just do `current_amis_for_mapping.update(amis_to_update.get(mapping_name, {}))` here?

tilne replied:
The reason it's done this way is to avoid the following:
```
>>> d1 = {'TopKey': {'SubkeyOne': 'SubvalueOne'}}
>>> d2 = {'TopKey': {'SubkeyTwo': 'SubvalueTwo'}}
>>> d1.update(d2)
>>> d1
{'TopKey': {'SubkeyTwo': 'SubvalueTwo'}}
```

In most cases I'd say this should be handled by a generic dict-merging function like the one we make heavy use of in the unit tests to easily express a config dict that differs from the defaults. But for one the structure of the mappings is known, and for another I like how spelling out the logic makes it more apparent how the function's affecting the structure that will eventually become the output of the script.

tilne commented on util/generate-ami-list.py:
```diff
@@ -37,29 +37,49 @@
+}
+
+
+def get_initialized_mappings_dicts():
```
my understanding is that we were moving away from that, especially for standalone scripts.

rexcsn replied:
I think we should still following the style guide here: https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles, and use `_` prefix for weak internal use indication in files that might be imported by others.

For this file I am ok with keeping as is because we are unlikely to use functions here elsewhere

tilne commented on util/get-ami-list.py:
```diff
@@ -26,42 +26,86 @@
-    repo.git.checkout(tag)
-
-    active_distro = None
+def parse_pre_280(repo_dir):
```
Done

tilne commented on util/get-ami-list.py:
```diff
@@ -26,42 +26,86 @@
+    return amis
+
 
+def parse_280_or_later(repo_dir):
```
done

tilne commented on util/generate-ami-list.py:
```diff
@@ -37,29 +37,49 @@
+        for region_name in regions:
+            if region_name in current_amis.get(mapping_name, []):
+                amis_json[mapping_name][region_name] = OrderedDict(
+                    sorted(current_amis.get(mapping_name).get(region_name).items())
```
Good idea. Done.

rexcsn replied:
Thanks for adding doc string everywhere.

I should have been clear, I was looking for a comment here like `# sort AMIs by OS` and/or a doc string `Return AMI list with OS in sorted order`. We sort by OS in this function and [L135](https://github.com/aws/aws-parallelcluster/pull/1836/files?file-filters%5B%5D=.py&file-filters%5B%5D=.txt#diff-b4b3aec78a3edee9ec9262bc418f1c4eR135) and we sort by region in `update_cfn_template`, which I think could be a little confusing, so just wanted to see a comment to call this out.

We want to write into CFN template an AMI json that is sorted by region and by OS in the end, maybe worth calling this out in a comment/doc string in `update_cfn_template` so we know the final format right away.

tilne commented:
> Dropped a couple of minor comments.
> I don't believe `ami_convert.py` and `get-ami-list.py` are used anymore, but it would be good to keep them updated since you have already done it.

@rexcsn Thanks for the review. I believe I've gotten to all of them. I'll consult the rest of the team regarding the amis.txt format question before merging.

tilne commented on amis.txt:
```diff
@@ -1,124 +1,132 @@
+## x86_64
```
Per internal meeting: leave format the same and eventually stop producing this file.

tilne commented:
@rexcsn After discussing the amis.txt format with the group this morning I believe I've addressed all your feedback.

Note also that after discussing it with @demartinofra I added a commit that deletes the two unused scripts.
